Fix "make distcheck"

wrp opened the pull request:
The 'distcheck' recipe should work, and it should execute the test suite.  This PR adds a wrapper script to invoke testprog and adds all the necessary source files to the distribution tarball so that the distcheck target can run.

This PR stacks on #113.  Since all of the autotool meta-data files are included in the repository, the diff appears significant, but this is just few lines in Makefile.am and the addition of the run-test script.

wrp committed:
Add "check" recipe to run testprog

Users of autoconfiscated packages expect to be able to run "make check"
to run the test suite.

wrp committed:
Include headers in the tarball

wrp committed:
Add utilities/{*.h,*.c} to tarball

esabol commented:
There's already a test for this in https://github.com/HEASARC/cfitsio/blob/develop/.github/workflows/ci.yml, so there's some redundancy here with your `run-test` script. If this were to be merged, the CI workflow should be modified to use your `run-test` script.

wrp committed:
Use check/distcheck targets in github workflow

esabol commented:
Thank you for adding the modifications to the ci.yml. 👍

wrp committed:
Change name from "run-test" to "run-testprog"

The script is strictly a wrapper around testprog, and the name
should reflect its use more accurately.

bryanirby commented:
Hi @wrp - I'm unable to merge this PR due to failed checks resulting from a missing "check-TESTS" target.  It's possible things got mangled during my review and attempted conflict resolution, but I see the same error when running 'make distcheck' on your add-check-target branch.  Are you able to comment on this issue?  Thanks.

wrp commented:
Rebased.   'make distcheck' works for me locally and is passing in the github runners.  Looks like the rebase restored the autoreconf changes to Makefile.in

bryanirby reviewed (approved):
I do note that 'make distcheck' seems fragile on some systems.  If a UID is greater than 2^21 - 1 (which mine are on our Macs) the configure decides that the UID is "not supported by ustar format", and this then results in a make failure with little information about the cause.  Closer inspection reveals that the UID setting causes the am__tar variable (intended for tar, gtar, gnutar) to be "none", which leads to the make distcheck error.
